src/contract_wrappers/modules/checkpoint/dividend_checkpoint_wrapper.ts

```diff
@@ -171,12 +180,12 @@ export default abstract class DividendCheckpointWrapper extends ModuleWrapper {
       created: bigNumberToDate(result[1]),
       maturity: bigNumberToDate(result[2]),
       expiry: bigNumberToDate(result[3]),
-      amount: result[4],
-      claimedAmount: result[5],
-      totalSupply: result[6],
+      amount: weiToValue(result[4], await (await this.securityTokenContract()).decimals.callAsync()),
```

Comment:
Amount is not expressed in SecurityTokens value, it is expressed in the ERC20 token for that particular dividend. It could be any other. 
You should call a protected method here to get the amount of decimals for the dividend and both checkpoint wrappers should implement it and return the right value.